backend/src/main/kotlin/ch/yvu/teststore/run/overview/RunOverviewService.kt

```diff
@@ -31,7 +33,12 @@ open class RunOverviewService @Autowired constructor(
         val runResult = extractRunResult(results);
         val totalDuration = results.map { it.durationMillis!! }.sum()
 
-        return RunOverview(run, RunStatistics(run.id, runResult, totalDuration, 0, 0, null, null))
+        val prevRun = runRepository.findLastRunBefore(run.testSuite!!, run.time!!)
+        val diff = resultDiffService.findDiff(prevRun?.id, run.id!!)
+        val numberPassed = diff[ResultDiffService.DiffCategory.NEW_PASSED]?.size ?: 0
```

Comment:
Since there are default values used for `numPassed` and `numFailed`, I would use these default values as well if any of the run attributes (testSuite, time, id) are null (instead of having to use `!!` and throw an exception if they don't exist).